sys/random/Makefile

```diff
@@ -11,4 +14,28 @@ ifneq (,$(filter prng_tinymt32,$(USEMODULE)))
   DIRS += tinymt32
 endif
 
+ifeq (,$(RIOT_CI_BUILD))
+  # random seed value newly generated on each build
+  RANDOM_SEED := $(shell $(RIOTTOOLS)/randhex/randhex.py 32)
+else
+  # hardcoded canary value to not break CI test cache
+  RANDOM_SEED := 0x1
+endif
+
+# This has to be kept in sync with sys/random/init.c
+ifneq (,$(filter puf_sram periph_hwrng periph_cpuid,$(USEMODULE)))
+  HAS_RNG = 1
+endif
+
+# If no run-time random number generation facility is available for
+# generating an initial seed, generate a random value at compile-time.
+# Also mark the file using it as PHONY to use a new seed on each build.
+ifndef HAS_RNG
+  $(warning Attention! The random module is using a compile-time seed. \
+    The same seed will be used for the same firmware image.)
```

Comment:
If someone has an idea how to word this better, i.e. explain in the shortest way possible what's going on, let me know.

Reply:
Actually, seeing this, I'm wondering if we should add this kind of message for the CPUID seeder too... Maybe a shared message if static seeds are involved? On the other had, we cannot add compile warnings everywhere :). What do you think?

Reply:
I think the CPUID seeder needs to be improved in general. I already laid out [my issues with the CPUID seeder and random ephemeral ports](https://github.com/RIOT-OS/RIOT/pull/13255#issuecomment-610332381). Maybe combine the CPUID seeder with the compile-time value or stop using it all together? I don't know. In any case, the CPUID seeder has the deficiency that one always ends up with the same seed on the same board. That should definitely be documented somewhere but maybe changes to the CPUID seeder should be discussed in a separate issue / PR?